corehq/apps/es/domains.py

```diff
@@ -55,9 +60,38 @@ class ElasticDomain(ElasticDocumentAdapter):
     def mapping(self):
         return get_adapter_mapping(self)
 
-    @classmethod
-    def from_python(cls, doc):
-        return from_dict_with_possible_id(doc)
+    def from_python(self, domain):
+        """
+        :param user: an instance of ``CouchUser`` or a user dict
+        :raises: ``TypeError`` user is none of the above types
+        """
+        from corehq.apps.domain.models import Domain
+        if isinstance(domain, Domain):
+            domain_dict = domain.to_json()
+        elif isinstance(domain, dict):
+            domain_dict = copy(domain)
+        else:
+            raise TypeError(f"Unkown type {type(domain)}")
+        return self.from_dict(domain_dict)
+
+    def from_dict(self, domain_dict):
```

Comment:
Why is this sometimes public and sometimes private?

Reply:
Thanks for catching it. It was missed when I was making this fn private in all the adapters.
Fixed in https://github.com/dimagi/commcare-hq/pull/32660/commits/e939243489fdacb4882ec4932cde054c48eca7d7